client/components/Facilitator/Components/Cohorts/index.js

```diff
@@ -0,0 +1,44 @@
+import { connect } from 'react-redux';
+import React from 'react';
+import PropTypes from 'prop-types';
+import { selectIndexRequest, selectCohortIds } from '@client/reducers/cohort';
+import { cohortRequestList } from '@client/actions/cohort';
+
+export class CohortIndex extends React.Component {
+    componentDidMount() {
+        if (this.props.requestIndex) {
+            this.props.requestIndex();
```

Comment:
I think this is an anti-pattern. At least it will cause an extra render. And another other components following the pattern will each cause an extra render. If they're in DidMount I don't think they'll be batched.

Actions should be called as a result of a user action. This request should be called by the event handler that will initiate the first React re-render.

I think we need a pattern for this to live as a redux middleware, where the store can quickly iterate without the more expensive component hierarchy and dom changes.

This is something that comes up in the Scratch GUI, where it changes the store as a result of a render. I'd like to help work on this if I cant.